piecrust/src/store/module_session.rs

```diff
@@ -10,13 +10,55 @@ use std::path::{Path, PathBuf};
 use std::sync::mpsc;
 use std::{io, mem};
 
+use crate::module::ModuleMetadata;
 use piecrust_uplink::ModuleId;
 
 use crate::store::{
-    compute_root, Bytecode, Call, Commit, Memory, Objectcode, Root,
-    BYTECODE_DIR, DIFF_EXTENSION, MEMORY_DIR, OBJECTCODE_EXTENSION,
+    compute_root, Bytecode, Call, Commit, Memory, Metadata, Objectcode, Root,
+    BYTECODE_DIR, DIFF_EXTENSION, MEMORY_DIR, METADATA_EXTENSION,
+    OBJECTCODE_EXTENSION,
 };
 
+#[derive(Debug, Clone)]
+pub struct ModuleData {
+    bytecode: Bytecode,
+    objectcode: Objectcode,
+    metadata: Metadata,
+    memory: Memory,
+}
```

Comment:
Since this is merely a piece of data that lives together, like the others before it, and it is also internal, I feel like making fields `pub` and removing accessors and `new` is fine. Maybe I'm wrong about this but it feels like we're trying to model this as an object when it's just plain data.